internal/k6runner/version/repository.go

```diff
@@ -0,0 +1,239 @@
+package version
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/rs/zerolog"
+)
+
+// Repository points to a collection of k6 binaries.
+type Repository struct {
```

Comment:
Do we need a constructor for this?

Reply:
Do we? I was thinking about instantiating the struct setting the properties. There's no magic to be done.